kubectl-plugin/test/e2e/kubectl_ray_log_test.go

```diff
@@ -0,0 +1,219 @@
+package e2e
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var requiredFileSet = map[string]string{
+	"stdout.log": "Ray runtime started",
+	"raylet.out": "Ray Event initialized for RAYLET",
+}
+
+var _ = Describe("Calling ray plugin `log` command on Ray Cluster", Ordered, func() {
+	It("succeed in retrieving all ray cluster logs", func() {
+		expectedDirPath := "./raycluster-sample"
+		expectedOutputStringFormat := `No output directory specified, creating dir under current directory using resource name\.\nCommand set to retrieve both head and worker node logs\.\nDownloading log for Ray Node raycluster-sample-head-\w+\nDownloading log for Ray Node raycluster-sample-small-group-worker-\w+`
+
+		cmd := exec.Command("kubectl", "ray", "log", "raycluster-sample", "--node-type", "all")
+		output, err := cmd.CombinedOutput()
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(strings.TrimSpace(string(output))).Should(MatchRegexp(expectedOutputStringFormat))
+
+		// Check that the log directory exists
+		logDirInfo, err := os.Stat(expectedDirPath)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(logDirInfo.IsDir()).To(BeTrue())
+
+		// Check the contents of the cluster directory
+		fileList, err := os.ReadDir(expectedDirPath)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(fileList).To(HaveLen(2))
+
+		for _, file := range fileList {
+			Expect(file.IsDir()).To(BeTrue())
+
+			// Check that the files exist and have correct content
+			logList, err := os.ReadDir(path.Join(expectedDirPath, file.Name()))
+			Expect(err).NotTo(HaveOccurred())
+
+			currentRequiredFileList := make(map[string]string)
+
+			for key, value := range requiredFileSet {
+				currentRequiredFileList[key] = value
+			}
+
+			for _, logFile := range logList {
+				if checkContent := currentRequiredFileList[logFile.Name()]; checkContent != "" {
+					delete(currentRequiredFileList, logFile.Name())
+
+					// read and check file content
+					fileContentByte, err := os.ReadFile(path.Join(expectedDirPath, file.Name(), logFile.Name()))
+					Expect(err).NotTo(HaveOccurred())
+
+					fileContent := string(fileContentByte)
+
+					Expect(fileContent).To(ContainSubstring(checkContent))
+				}
+			}
+
+			Expect(currentRequiredFileList).To(BeEmpty())
+		}
+
+		// Cleanup
+		err = os.RemoveAll(expectedDirPath)
```

Comment:
Should we put this in a cleanup function so that it will run even if the test is canceled during execution?

Reply:
Added a "AfterEach" block.